native/src/overlay/nested.rs

```diff
@@ -0,0 +1,359 @@
+                    );
+                }
+            }
+        }
+
+        self.overlay.with_element_mut(|element| {
+            recurse(element, layout, renderer, theme, style, cursor_position);
+        })
+    }
+
+    fn operate(
+        &mut self,
+        layout: Layout<'_>,
+        renderer: &Renderer,
+        operation: &mut dyn widget::Operation<Message>,
+    ) {
+        fn recurse<Message, Renderer>(
+            element: &mut overlay::Element<'_, Message, Renderer>,
+            layout: Layout<'_>,
+            renderer: &Renderer,
+            operation: &mut dyn widget::Operation<Message>,
+        ) where
+            Renderer: crate::Renderer,
+        {
+            let mut layouts = layout.children();
+
+            if let Some(layout) = layouts.next() {
+                element.operate(layout, renderer, operation);
+
+                if let Some((mut nested, nested_layout)) =
+                    element.overlay(layout, renderer).zip(layouts.next())
+                {
+                    recurse(&mut nested, nested_layout, renderer, operation);
+                }
+            }
+        }
+
+        recurse(self.overlay.0.get_mut(), layout, renderer, operation)
+    }
+
+    fn on_event(
+        &mut self,
+        event: Event,
+        layout: Layout<'_>,
+        cursor_position: Point,
+        renderer: &Renderer,
+        clipboard: &mut dyn Clipboard,
+        shell: &mut Shell<'_, Message>,
+    ) -> event::Status {
+        fn recurse<Message, Renderer>(
+            element: &mut overlay::Element<'_, Message, Renderer>,
+            layout: Layout<'_>,
+            event: Event,
+            cursor_position: Point,
+            renderer: &Renderer,
+            clipboard: &mut dyn Clipboard,
+            shell: &mut Shell<'_, Message>,
+        ) -> event::Status
+        where
+            Renderer: crate::Renderer,
+        {
+            let mut layouts = layout.children();
+
+            if let Some(layout) = layouts.next() {
+                let status = if let Some((mut nested, nested_layout)) =
+                    element.overlay(layout, renderer).zip(layouts.next())
+                {
+                    recurse(
+                        &mut nested,
+                        nested_layout,
+                        event.clone(),
+                        cursor_position,
+                        renderer,
+                        clipboard,
+                        shell,
+                    )
+                } else {
+                    event::Status::Ignored
+                };
+
+                if matches!(status, event::Status::Ignored) {
+                    element.on_event(
+                        event,
+                        layout,
+                        cursor_position,
+                        renderer,
+                        clipboard,
+                        shell,
+                    )
+                } else {
+                    status
+                }
```

Comment:
Still need to encode cursor availability here when the overlay a layer higher `is_over`.

I was going to, but there was an issue in the `modal` example because of how `pick_list` handles mouse clicks on it's overlay menu. It doesn't capture the event when clicked in the overlay, and instead lets the base widget handle "closing" and publishing to shell the selection. 

This is an issue because then the nested pick_list (higher layer) reports `is_over`, but doesn't capture the event. So if we set cursor to `(-1, -1)`, the "modal" overlay will think it's clicked outside and close.

I think we need to refactor `pick_list` to "capture" the mouse click in the overlay menu.

Reply:
I've resolved this with a617f3a and 4d439e4.

You can compare the behavior prior to a617f3a on `modal` example. If you select the middle picklist option over the text input, the text input focuses after selecting the option.

After a617f3a, the undesired behavior mentioned above occurs (modal closes when clicking outside menu bounds / onto modal).

After 4d439e4, the picklist menu now properly captures the status of the event in the overlay when an option is selected. Everything now works as expected.